components/Feature/VulnerabilitiesGrid/index.js

```diff
@@ -125,148 +134,191 @@ const VulnerabilitiesGrid = ({ resources, onUpdate }) => {
     });
   };
 
+  const setAllExpandedGroups = () => {
+    let expandedGroupsState = false;
+    let values = {};
+    groups.forEach(({ id }) => {
+      if (!expandedGroups[id] || expandedGroups[id] === false) {
+        expandedGroupsState = true;
+        return;
+      }
+    });
+    groups.forEach(({ id }) => {
+      values = Object.assign(values, { [id]: expandedGroupsState });
+    });
+    setExpandedGroups(values);
+  };
```

Comment:
i think we can cut this down a lot. eg.

```
const setAllExpandedGroups = () => {
const hasExpanded = Object.values(expanded).includes(true);
setExpandedGroups(groups.reduce((acc, curr) => {acc[curr.id] = !hasExpanded; return acc}, {}));
}

Reply:
Didn't change that just yet, doesn't seem to work right :'(

Reply:
If I get the example you gave right: if at least one of the groups is expanded, it will close all of them, whereas we want to open them if at least one of them is closed (or undefined). So we want to go through all of the IDs and check them against expandedGroups, cause expandedGroups might be empty etc.

Reply:
So if at least one is open we want to open all? Unless they're all open so we close?

``` 
const expandedItems = Object.values(expanded);
const allExpanded = expandedItems.includes(true) && expandedItems.length === groups.length;
setExpandedGroups(groups.reduce((acc, curr) => {acc[curr.id] = !allExpanded; return acc}, {}));
```

Reply:
OK so if only one is expanded we want to expand all. If all expanded we want to close? Something _like_ this
```
const expandedItems = Object.values(expanded);
const allExpanded = expandedItems.every(Boolean) && expandedItems.length === groups.length;
setExpandedGroups(groups.reduce((acc, curr) => {acc[curr.id] = !allExpanded; return acc}, {}));